### ISSUE TITLE
bookmarklet can not annotate github - should add a warning to explain

### ISSUE THREAD

**Issue by drewfrank** [hypothesis/h#368]
I came here to post another issue but found I was unable to actually preview or submit it. The hypothesis Chrome extension is somehow interfering with Github's site.

Steps to reproduce:
1. Enable the Chrome extension.
2. Click the "New Issue" button on Github
3. Type a title and some content.

The submit button will remain grayed out and previewing will not work.

**Comment by glassdimly** [comment]
Hmm. I just installed the bookmark on FF, went to github.com to annotate, and I got this error (`Content Security Policy: The page’s settings blocked the loading of a resource at inline (“script-src”).`). This thread is from 2014, but perhaps it's time to revisit this?

The fix here might be to release a very small FF plugin which specifies a CSP override. [This is possible](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy) (correct me if I'm wrong).

Other fix ideas...?

As many websites now implement a restrictive CSP, this seems a more urgent issue than it did in 2014. Is there any motion around developing such a small plugin for FF to fix this issue? I would think many privacy advocates and scholars use FF: Hypthesis' target audience.

**Comment by AlexHankins** [comment]
I just noticed this on github.  I use the Firefox bookmarklet.  It took a lot of head-scratching before I tried checking the developer console for errors...

**Comment by glassdimly** [comment]
Yes, for normal users with no understanding of CSP this is just erratically broken.

**Comment by diegodlh** [comment]
@glassdimly, @AlexHankins, I was in the same situation. Until a Firefox addon is released I worked around it by automatically adding hypothes.is to the list of allowed script, style, font and sources in CSP.  

To do so, I'm using the [Header Editor addon](https://addons.mozilla.org/es/firefox/addon/header-editor/) for Firefox, with a custom function I wrote [here](https://github.com/diegodlh/h.csp/blob/master/header_editor) (more information about writing custom functions for Header Editor [here](https://github.com/FirefoxBar/HeaderEditor/wiki/Custom-function)). You can simply import the rule entering this url into the 'Export and Import' section of the addon: https://raw.githubusercontent.com/diegodlh/h.csp/master/hypothesis.json
In short, what it does is:
1. it reads the CSP response header sent by the webserver,
2. if x-src exists (where x is one of 'script', 'style', 'frame', 'font', and 'frame'), adds hypothes.is domain to it,
3. if x-src does not exist and default-src includes 'none', add a x-src policy with the hypothes.is domain as source.

Probably this can be improved, but it helped me work around this until a better fix is found.

**Comment by diegodlh** [comment]
I felt that writing a short guide about how to use Hypothesis in Firefox could be helpful for others. I included the CSP workaround in step 2: https://medium.com/@delahera/hypothesis-web-annotation-mobile-358db1f9ec46
